### PR TITLE
Support jQuery 3

### PR DESCRIPTION
**Pull request by rdennis**
Adds support for jQuery 3 as reported in issue #78.

I am not familiar with CoffeeScript, so please make this more Coffee correct if I have butchered it.

### CODE REVIEW

**Comment by rvera** [comment]
Thanks @rdennis, I'm a bit swamped but will take a look at this over the week.

File: source/coffee/image-picker.coffee [hunk 1]
```diff
@@ -26,14 +26,11 @@ sanitized_options = (opts) ->
...
-  if (!a || !b) || (a.length != b.length)
-    return false
-  b = b[..]
-  for x in a
-    i = b.indexOf x
-    if i < 0
-      return false
-    b.splice(i, 1)
+  return false if (!a || !b) || (a.length != b.length)
+  a.sort()
```
Comment (rdennis) [inline source/coffee/image-picker.coffee]: I think that sorting the arrays and comparing each element sequentially will be faster than using `indexOf`, but I have not done any testing to validate my hypothesis. For small data sets the performance difference will almost certainly not be noticeable.

File: source/coffee/image-picker.coffee [hunk 2]
```diff
@@ -26,14 +26,11 @@ sanitized_options = (opts) ->
...
 both_array_are_equal = (a,b) ->
-  if (!a || !b) || (a.length != b.length)
-    return false
-  b = b[..]
-  for x in a
-    i = b.indexOf x
-    if i < 0
-      return false
-    b.splice(i, 1)
+  return false if (!a || !b) || (a.length != b.length)
```
Comment (rdennis) [inline source/coffee/image-picker.coffee]: I think that sorting the arrays and comparing each element sequentially will be faster than using `indexOf`, but I have not done any testing to validate my hypothesis. For small data sets the performance difference will almost certainly not be noticeable.

**Comment by coneybeare** [comment]
Fix works for me.

File: source/coffee/image-picker.coffee [hunk 3]
```diff
@@ -26,14 +26,11 @@ sanitized_options = (opts) ->
   jQuery.extend(default_options, opts)
 
 both_array_are_equal = (a,b) ->
```
Comment (rdennis) [inline source/coffee/image-picker.coffee]: I think that sorting the arrays and comparing each element sequentially will be faster than using `indexOf`, but I have not done any testing to validate my hypothesis. For small data sets the performance difference will almost certainly not be noticeable.

File: source/coffee/image-picker.coffee (hunk 2 above)
Comment (rvera) [inline source/coffee/image-picker.coffee]: Dunno why but just got an email regarding this!

I think one downside of this is that `.sort` is an inplace operation. ie, this modifies the original arrays. Need to check the code to see if this has any impact on the rest of the code.

```
var a = [3,2,1]
var b = [3,2,4]
function sortArrays(a, b) {
    a.sort()
    b.sort()
}

console.log(a,b)
>>> [3, 2, 1] [3, 2, 4]
sortArrays(a,b)
console.log(a,b)
>>> [1, 2, 3] [2, 3, 4]
```

Comment (rdennis) [inline source/coffee/image-picker.coffee]: Ah good point. We could simply clone the arrays. 

```
var a = [3,2,1]
var b = [3,2,4]
function sortArrays(a, b) {
    a = a.slice(0)
    b = b.slice(0)
    a.sort()
    b.sort()
    console.log(a, b)
}

console.log(a, b)
>>>> [3, 2, 1] [3, 2, 4]
sortArrays(a, b)
>>>> [1, 2, 3] [2, 3, 4]
console.log(a, b)
>>>> [3, 2, 1] [3, 2, 4]
```

If we know that the arrays are always already sorted we could skip sorting in this function.

**Comment by rdennis** [comment]
I made the change to clone `a` and `b`.

**Comment by rvera** [comment]
Excellent work! Thanks for the patience.

**Pull request merged**